core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -151,6 +160,13 @@ public void renameAllTempFiles() throws IOException {
         store.renameTempFilesSafe(tempFileNames);
     }
 
+    public void close() {
```

Comment:
I'm a bit concerned with exposing this close() as public is - someone may think this is the way to finish / clean up (which you should call `cancel` or `markAsDone`). How about calling this `closeAndStartNew` and make it return the new RecoveryTarget ? this also means we can make the copy constructor private, which is a good thing?